repose-aggregator/core/core-lib/src/main/java/org/openrepose/powerfilter/PowerFilter.java

```diff
@@ -383,6 +384,21 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
             mutableHttpResponse.sendError(HttpServletResponse.SC_BAD_GATEWAY, "Error processing request");
             mutableHttpResponse.setLastException(ex);
         } finally {
+            try {
+                boolean isValidMethod = false;
+                for (HttpComponentFactory hcf : HttpComponentFactory.values()) {
+                    if (hcf.name().equalsIgnoreCase(mutableHttpRequest.getMethod())) {
```

Comment:
Arrays.asList(yourArray).contains(yourValue)

No need for the boolean variable. Just use a conditional based on the line above. If the method is not supported, reject the request.